esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,914 @@
+        }
+        return xpathset;
+    }
+};
+
+
+class CEsdlTransformOperationSetValue : public CEsdlTransformOperationBase
+{
+protected:
+    Owned<ICompiledXpath> m_select;
+    Owned<ICompiledXpath> m_xpath_target;
+    StringAttr m_target;
+    StringAttr m_traceName;
+
+public:
+    CEsdlTransformOperationSetValue(IPropertyTree *tree, const StringBuffer &prefix) : CEsdlTransformOperationBase(tree, prefix)
+    {
+        m_traceName.set(tree->queryProp("@name"));
+
+        if (isEmptyString(tree->queryProp("@target")) && isEmptyString(tree->queryProp("@xpath_target")))
+        {
+            VStringBuffer msg(" %s without target", tagname.str());
+            esdlOperationError("Custom transform: ", msg.str(), m_traceName.str(), !ignoreCodingErrors);
+        }
+
+        const char *select = tree->queryProp("@select");
+        if (!select)
+            select = tree->queryProp("@value");
+        if (isEmptyString(select))
+        {
+            VStringBuffer msg(" %s without select", tagname.str());
+            esdlOperationError("Custom transform: ", msg.str(), m_traceName.str(), !ignoreCodingErrors);
+        }
+
+        m_select.setown(compileXpath(select));
+        if (tree->hasProp("@xpath_target"))
+            m_xpath_target.setown(compileXpath(tree->queryProp("@xpath_target")));
+        else if (tree->hasProp("@target"))
+            m_target.set(tree->queryProp("@target"));
+    }
+
+    virtual void toDBGLog() override
+    {
+#if defined(_DEBUG)
+        DBGLOG(">%s> %s(%s, select('%s'))", m_traceName.str(), tagname.str(), m_target.str(), m_select->getXpath());
+#endif
+    }
+
+    virtual ~CEsdlTransformOperationSetValue(){}
+
+    virtual bool process(IEspContext * context, IPropertyTree *request, IXpathContext * xpathContext) override
+    {
+        if (!xpathContext)
+            throw MakeStringException(-1, "Could not process custom transform (xpathcontext == null)");
+        if (!request)
+            return false;
+        if ((!m_xpath_target && m_target.isEmpty()) || !m_select)
+            return false; //only here if "optional" backward compatible support for now (optional syntax errors aren't actually helpful
+        try
+        {
+            StringBuffer value;
+            xpathContext->evaluateAsString(m_select, value);
+            return doSet(xpathContext, request, value);
+        }
+        catch (IException* e)
+        {
+            StringBuffer msg;
+            esdlOperationError("Custom transform: ", e->errorMessage(msg), m_traceName, !ignoreCodingErrors);
+            e->Release();
+        }
+        catch (...)
+        {
+            esdlOperationError("Custom transform: ", "could not process", m_traceName, !ignoreCodingErrors);
+        }
+        return false;
+    }
+
+    const char *getTargetPath(IXpathContext * xpathContext, StringBuffer &s)
+    {
+        if (m_xpath_target)
+        {
+            xpathContext->evaluateAsString(m_xpath_target, s);
+            return s;
```

Comment:
If the string turns out to be an invalid ptree xpath the setProp should throw a fairly understandable exception for finding the script bug.
My test case shows how valuable supporting this can be.  In that case in the for loop I'm able to build an xpath to the correct target destination for each iteration through the loop.